File: general-solving/leetcode/0104_maximum-depth-of-binary-tree.cpp
/***************************************************************************
*   Problem Name:   Determines if given string has some properties
*   Problem URL :   https://leetcode.com/problems/maximum-depth-of-binary-tree/
                    related: https://www.hackerrank.com/challenges/tree-height-of-a-binary-tree
*   Domain      :   data-structures/trees
*   DateTime    :   August 7, 2015
*   Desc        :   Recursion on Binary Tree
*                   
*
*   Complexity  :   O(n), .008s
*   Author      :   Atiq Rahman
*   Status      :   Accepted
***************************************************************************/

int height(node * root)
{
  if (root == NULL)
      return 0;
  return std::max(height(root->left)+1, height(root->right)+1);
}

File: general-solving/leetcode/0104_maximum-depth-of-binary-tree.cs
/***************************************************************************************************
* Title : Determines if given string has some properties
* URL   : https://leetcode.com/problems/maximum-depth-of-binary-tree/
* rel   : https://www.hackerrank.com/challenges/tree-height-of-a-binary-tree
*   http://www.lintcode.com/en/problem/maximum-depth-of-binary-tree/
*   
* Date  : 2015-08-07
* Notes : Recursion on Binary Tree
*   Maximum depth is the max of depth from the children
*   Domain- data-structures/trees
* Comp  : O(n), .008s
* Author: Atiq Rahman
* Status: Accepted
* meta  : tag-binary-tree, tag-recursion, tag-leetcode-easy
***************************************************************************************************/
public class Solution
{
  public int MaxDepth(TreeNode root) {
    if (root == null)
      return 0;
    return Math.Max(MaxDepth(root.left)+1, MaxDepth(root.right)+1);
  }
}
